src/scan.js

```diff
@@ -21,7 +23,12 @@ var _curry3 = require('./internal/_curry3');
  *      var factorials = R.scan(R.multiply, 1, numbers); //=> [1, 1, 2, 6, 24]
  */
 module.exports = _curry3(function scan(fn, acc, list) {
-  var idx = 0, len = list.length, result = [acc];
+  if (!_isArray(list) && typeof list.scan === 'function') {
```

Comment:
I suppose not, if people are modifying the array prototype they might be doing so knowingly. I know we intercept it this way for reduce